stage #3/java/module #2. REST API Basics/task02/repository/src/test/java/by/epam/esm/RepositoryConfig.java

```diff
@@ -0,0 +1,75 @@
+package by.epam.esm;
+
+
+import by.epam.esm.dao.CertificateSqlQueryAssemblerImpl;
+import by.epam.esm.dao.CrdTagDao;
+import by.epam.esm.dao.CrudGiftCertificateDao;
+import by.epam.esm.dao.impl.GiftCertificateDaoImpl;
+import by.epam.esm.dao.impl.TagDaoImpl;
+import by.epam.esm.dao.mapper.GiftCertificateRowMapper;
+import by.epam.esm.dao.mapper.TagRowMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+
+import javax.sql.DataSource;
+
+@Configuration
+public class RepositoryConfig {
+    private static final String DB_SCRIPT = "classpath:DBScript.sql";
+    @Autowired
+    private DataSource dataSource;
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+    @Autowired
+    private TagRowMapper tagRowMapper;
+    @Autowired
+    private GiftCertificateRowMapper giftCertificateRowMapper;
+    @Autowired
+    private CertificateSqlQueryAssemblerImpl sqlQuery;
+    @Autowired
```

Comment:
So, Are you sure you need to autowire beans and then create them through @Bean annotation?
You use both @Component and @Bean approaches, please read more about how it works